### PR TITLE
hard-enable the defmt Cargo feature

### PR DESCRIPTION
**Pull request by japaric**
but keep it around so `cargo install --git $URL --features defmt` continues to work
with this, probe-run will always include defmt support (behind the --defmt command line flag)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -28,7 +28,8 @@ signal-hook = "0.1.16"
 structopt = "0.3.15"
 
 [features]
-defmt = ["defmt-elf2table", "defmt-decoder"]
+# this feature does nothing and it's kept for backwards compatibility; it will be removed in the next semver bump
```
Comment (Lotterleben) [inline Cargo.toml:31]: nit: would it make sense to note this down as a low-prio issue as a reminder or is that overkill?

Comment (japaric) [reply Cargo.toml:31]: hmm, actually I think adding a TODO comment to this Cargo.toml just above the `version = ` field would be more visible. (That way whoever bumps the minor version next will be reminded to remove this feature)

Comment (Lotterleben) [reply Cargo.toml:31]: ha! yeah that makes sense and causes less clutter

**Review by Lotterleben — APPROVED** [review]
Approving to get the PR out of your way but I'm not 100% confident I understand all the implications yet

**Pull request merged**